### PR TITLE
2# PACJENT - Pytania i szablony ankiet 

### PR DESCRIPTION
**Pull request by MacWar**
Main:
- Created `Questionnaire` component that retrieves data from patient
- Created `questionnaire` endpoint
- Created `questionnaireDTO ` transfer model
- Created `quetionnaire` model
- Created database migration

Additional:
- Added basic style to `Questionnaire` component

### CODE REVIEW

**Comment by Hoodster** [comment]
All dlls and other build files should be removed from pull request

File: MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            alert("Proszę udzielić odpowiedzi na wszystkie pytania aby przejść dalej")
+        }
+        else {
+            this.postQuestionnarieToApi();
+        }
+    }
+
+    render() {
+     
+    const displayQuestions = this.questions.map( (question, index) => {
```
Comment (Hoodster) [inline MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:162]:
```suggestion
    const displayQuestions = this.questions.map((question, index) => {
```

File: MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        let stateValue = this.state[Object.keys(this.state)[index]]
+        return(
+            <Question key={question.id} answer={stateValue} question={question.question} handleCheboxChange={this.handleCheboxChange} questionId={question.id}/>
```
Comment (Hoodster) [inline MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:165]: Typo in `handleCheckboxChange`

File: MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            question : "Czy wykonywane były naświetlania lub/i podawane leki cytostatyczne? (leki stosowane w leczeniu chorób nowotworowych) "
+        },
+        {
+            id : 21,
+            question : "Czy był wykonywany lub planowany jest przeszczep narządów? "
+        }
+
+    ]
+    
+    postQuestionnarieToApi = () => {
```
Comment (Hoodster) [inline MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:118]: Typo

File: MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    handleCheboxChange = (questionId)=> (event)=> {
+        let questionStateKey = `question${questionId}`
+        
+        if(event.target.checked)
+            this.setState ({
+                [questionStateKey] : event.target.name
+            })
+        else if(!event.target.checked){
```
[the remark this answers is not in the document]
Comment (Hoodster) [reply MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:134]: Ok I noticed that there's "i don't know" option. Then I suggest to make enum like
```
Yes,
No,
NotKnown
```

[the remark this answers is not in the document]
Comment (Hoodster) [reply MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:134]: Also, please change what you retrieve from event. `event.target.name` is misunderstanding, it's better to pass value from input.

File: MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    postQuestionnarieToApi = () => {
+        axios.post(
+            'http://localhost:17921/api/account/questionnaire',this.state
+        ) .then(resp => {
+            console.log(resp);
+            console.log(resp.data);
+        })
+    }
+
+    handleCheboxChange = (questionId)=> (event)=> {
```
[the remark this answers is not in the document]
Comment (MacWar) [reply MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:127]: The only way I know to pass the parameter and the event at the same time. Is it incorrect?

[the remark this answers is not in the document]
Comment (Hoodster) [reply MedSystem/Frontend/client/src/components/Questionnaire/Questionnaire.js:127]: Yes, it's correct. I forgot there's no other way. I close this thread then.

File: MedSystem/Frontend/client/src/App.js [hunk 6]
```diff
@@ -7,7 +6,7 @@ import Registration from './views/Registration';
 
 function App() {
   return (
-    <div className="App disp-center disp-vertical">
+    <div className="App ">
```
Comment (Namtrak) [inline MedSystem/Frontend/client/src/App.js:9]: The questionnaire should be on a separate route

**Pull request merged**